src/app.ts

```diff
@@ -40,10 +41,10 @@ export class App {
     // Swagger docs
     SwaggerDocs(app);
 
-    app.useGlobalInterceptors(new TransformInterceptor());
-    app.useGlobalInterceptors(new LoggerErrorInterceptor());
     app.useLogger(app.get(Logger));
-
+    app.useGlobalInterceptors(new LoggerErrorInterceptor());
+    app.useGlobalGuards(new ApiKeyAuthGuard());
+    app.useGlobalInterceptors(new TransformInterceptor());
```

Comment:
Does the order matter here?

If it does, then is it okay that we swapped the `TransformInterceptor` and `LoggerErrorInterceptor`?
If it does not, then it's slightly strange that we have the guard between the two interceptors

Reply:
Yes, the order does matter. 
I've updated the order now to say: if users don't have the API Key, then don't proceed with anything else

Reply:
Ah interesting - that does mean we don't log anywhere that we had an unauthenticated request. The logging we have at the moment just says 
> [18:07:36.933] INFO (7336): request completed <info about the request and response>

If you look through the info about the request and response then you might be able to spot where it says `"res": { "statusCode": 401` but it's hard to see.

Maybe we should open a ticket to improve the logging for this? Or do we think the monitoring at the infrastructure level will be good enough at spotting this?

Reply:
I've tried to swap their order and I'm getting the same type of messages: 
```
request completed {"req":{"id":1,"method":"GET","url":"/api.." }
```

it will be worth investigating it further in a separate ticket. although it would be good to know what we do expect to see in the response?

Reply:
Thanks Gabriel. I've raised APIM-161 in the backlog. We can decide as part of that ticket exactly _what_ we should do, if anything